src/main/java/frc/robot/Robot.java

```diff
@@ -94,4 +153,157 @@ public void teleopPeriodic() {
   @Override
   public void testPeriodic() {
   }
+
+  /**
+   * This function drives the robot at a certain speed.
+   */
+  public void driveSpeed() {
+    if (!colorWheelMode) {
+      double rawAxis1 = driveController.getRawAxis(1);
+      double rawAxis4 = driveController.getRawAxis(4);
+
+      // Enabling slow speed toggle on the joystick for drive speed
+      if (driveController.getRawAxis(2) > 0.5) {
+        slowToggle = true;
+      } else {
+        slowToggle = false;
+      }
+
+      // Enabling high speed toggle on the joystick for drive speed
+      if (driveController.getRawAxis(3) > 0.5) {
+        highToggle = true;
+      } else {
+        highToggle = false;
+      }
+
+      // Setting robot drive speed
+      if (slowToggle) {
+        differentialDrive.arcadeDrive(rawAxis1 * slowSpeed, rawAxis4 * slowSpeed);
+      } else if (highToggle) {
+        differentialDrive.arcadeDrive(rawAxis1 * highSpeed, rawAxis4 * highSpeed);
+      } else {
+        differentialDrive.arcadeDrive(rawAxis1, rawAxis4);
+      }
+    }
+  }
+
+  /*
+   * This function is used to select the color for the color wheel.
+   */
+  public void selectColor() {
+    // Button five (lb) is used as the button to activate color selection.
+    boolean lb = manipulateController.getRawButton(5);
+    // Button one (A) selects the color green.
+    boolean a = manipulateController.getRawButton(1);
+    // Button two (B) selects the color red.
+    boolean b = manipulateController.getRawButton(2);
+    // Button three (X) selects the color blue.
+    boolean x = manipulateController.getRawButton(3);
+    // Button four (Y) selects the color yellow.
+    boolean y = manipulateController.getRawButton(4);
+
+    if (lb) {
+      if (colorWheelMode) {
+        colorWheelMode = false;
+      } else if (!colorWheelMode) {
+        colorWheelMode = true;
+      }
+    }
+
+    if (colorWheelMode) {
+      if (a) {
+        colorSensorDetectGreen = true;
+      } else if (b || x || y) {
+        colorSensorDetectGreen = false;
+      }
+      if (b) {
+        colorSensorDetectRed = true;
+      } else if (a || x || y) {
+        colorSensorDetectRed = false;
+      }
+      if (x) {
+        colorSensorDetectBlue = true;
+      } else if (a || b || y) {
+        colorSensorDetectBlue = false;
+      }
+      if (y) {
+        colorSensorDetectYellow = true;
+      } else if (a || b || x) {
+        colorSensorDetectYellow = false;
+      }
+    }
+  }
+
+  /**
+   * This function sets how many spins are required on the color wheel.
+   */
+  public void setColorWheelSpinAmount() {
```

Comment:
This method is not called. Do you intend on calling this alongside `selectColor` in `teleopPeriodic`? If so, this should be renamed to `selectColorWheelSpinAmount`, and be called there. I say to change the `set` to `select` not only for analogy with `selectColor`, but because `set` falsely implies that this is a [setter method](https://en.wikipedia.org/w/index.php?title=Setter_method).

Reply:
Yes, this is now called alongside `selectColor` in `teleopPeriodic`. It has been renamed to `selectColorWheelSpinAmount` as well.